File: Tracks/typescript/anagram/anagram.ts
export default class Anagram {
  private readonly baseWord: string;

  constructor(baseWord: string) {
    this.baseWord = baseWord.toLowerCase();
  }

  matches = (...words: string[]): string[] =>
    words.filter(
      word =>
        word.length === this.baseWord.length &&
        word.toLowerCase() !== this.baseWord &&
        [...word.toLowerCase()].sort().join() ===
          [...this.baseWord].sort().join()
    );
}

File: Tracks/csharp/anagram/Anagram.cs
using System;
using System.Linq;

public class Anagram
{
    private readonly string baseWord;

    public Anagram(string baseWord) => this.baseWord = baseWord.ToLower();

    private bool IsAnagram(string word)
    {
        word = word.ToLower();
        return (word == baseWord || word.Length != baseWord.Length) 
            ? false : word.OrderBy(letter => letter).SequenceEqual(baseWord.OrderBy(letter => letter));
    }

    public string[] FindAnagrams(string[] potentialMatches) => potentialMatches.Where(IsAnagram).ToArray();
}